libs/store/src/lib/infrastructure/persistence/utils/entity-options.service.ts

```diff
@@ -46,10 +48,10 @@ export class DefaultEntityMetaOptionsService implements EntityMetaOptionsService
         return options;
     }
 
-    getEntityMetadata<T>(entityOrName: string | Type<T>): EntityMetaOptions {
-        const options =
+    getEntityMetadata<T extends BaseEntity>(entityOrName: string | EntityType<T>): EntityMetaOptions<T> {
+        const options: EntityMetaOptions<T>  =
             typeof entityOrName === 'string'
-                ? getEntityMetadataByEntityName(entityOrName)
+                ? getEntityMetadataByEntityName<T>(entityOrName)
```

Comment:
I'm seeing the following error when running `ng test store`:
```
libs/store/src/lib/infrastructure/persistence/utils/entity-options.service.ts:54:19 - error TS2352: Conversion of type 'EntityMetaOptions<BaseEntity>' to type 'EntityMetaOptions<T>' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Types of property 'chainingPolicy' are incompatible.
    Type 'ChainingPolicy<BaseEntity>' is not comparable to type 'ChainingPolicy<T>'.
      Type 'BaseEntity' is not comparable to type 'T'.
        'BaseEntity' is assignable to the constraint of type 'T', but 'T' could be instantiated with a different subtype of constraint 'BaseEntity'.

54                 ? getEntityMetadataByEntityName(entityOrName) as EntityMetaOptions<T>
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
That's strange I don't see such error.
I'm using the workspace typescript version 3.9.7
Could you try `npx ng test store` command to be sure that you are using the same